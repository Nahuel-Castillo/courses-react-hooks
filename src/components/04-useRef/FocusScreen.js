import React, { useRef } from 'react'
import { useForm } from '../../hooks/useForm';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {

    const inputRef = useRef();

    const [ values, handleInputChange ] = useForm(); 
    const { name } = values;

    const handleClick = () => {
        inputRef.current.select();
    };

    return (
        <div>
            <h2> FocusScreen</h2>

            <input 
                ref= { inputRef }
                name="name"
                className="form-control"
                placeholder="Nombre"
                value={ name }
                onChange={ handleInputChange }
            />

            <button 
                className="btn btn-outline-primary mt-4"
                onClick={ handleClick}
            >
                Focus
            </button>
        </div>
    )
}
