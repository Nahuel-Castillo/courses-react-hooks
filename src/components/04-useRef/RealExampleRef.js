import React, { useEffect, useRef, useState } from 'react';
import '../02-useEffect/effects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h2> RealExampleRef </h2>
            <hr/>

            { show && <MultipleCustomHooks /> }
            <button
                className="btn btn-primary mt-4"
                onClick={ () => { setShow(!show) }}
            >
                Show/Hide Quote
            </button>
        </div>
    )
}
