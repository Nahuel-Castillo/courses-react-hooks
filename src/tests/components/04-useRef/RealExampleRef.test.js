import React from 'react';
import { shallow } from 'enzyme';

import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";

describe('Test in <RealExampleRef/>', () => {
    
    test('should show correctly', () => {
        
        const wrapper = shallow(<RealExampleRef />);
        
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should show <MultipleCustomHooks/>', () => {
        
        const wrapper = shallow(<RealExampleRef />);

        wrapper.find('button').simulate('click');

        const mch = wrapper.find('MultipleCustomHooks');

        expect( mch.exists() ).toBe( true );
    });
});