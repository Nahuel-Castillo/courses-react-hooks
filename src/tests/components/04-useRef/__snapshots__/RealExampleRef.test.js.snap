// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <RealExampleRef/> should show correctly 1`] = `
<div>
  <h2>
     RealExampleRef 
  </h2>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show/Hide Quote
  </button>
</div>
`;
